pkg/manager/task_scan.go

```diff
@@ -48,19 +48,33 @@ func (t *ScanTask) scanGallery() {
 	qb := models.NewGalleryQueryBuilder()
 	gallery, _ := qb.FindByPath(t.FilePath)
 
+	fileModTime, err := t.getFileModTime()
+	if err != nil {
+		logger.Error(err.Error())
+		return
+	}
+
 	if gallery != nil {
 		// We already have this item in the database, keep going
 
-		// if the mod time of the zip file is newer than that of the associated
-		// gallery, then recalculate the checksum
-		modified, err := t.isFileModified(gallery.UpdatedAt)
-		if err != nil {
-			logger.Error(err.Error())
-			return
+		// if file mod time is not set, set it now
+		if !gallery.FileModTime.Valid {
+			t.updateFileModTime(gallery.ID, fileModTime, &qb)
```

Comment:
This is not enough.
While the existing gallery's mod time gets updated the images included in the zip file don't get their mod times updated and end up with a null file_mod_time after the scan.